packages/ptokens-entities/src/ptokens-asset.ts

```diff
@@ -42,71 +53,65 @@ export type AssetFees = {
   basisPoints: NativeToXBasisPoints | HostToXBasisPoints
 }
 
-export type AssetInfo = {
+type pTokenAsset = {
+  /** The name of the asset. */
+  name: string
+  /** The chain ID of the asset's blockchain. */
+  chainId: string
+  /** Asset symbol */
+  symbol: string
+  /** pToken address. */
+  pTokenAddress: string
+  /** Token's decimals. */
+  decimals: number
+  /** Underlying asset information */
+  underlyingAsset: NativeAsset
+}
+
+type NativeAsset = {
```

Comment:
Looks like `NativeAsset` can be avoided by adding a boolean to `pTokenAsset` telling if it's native or not?

Reply:
In that case I would keep `AssetInfo`, since it's more general

Reply:
It is needed to keep track of what is the underlying asset in case you are dealing with a pToken.